### ISSUE TITLE
Put DKAN on drupal.org

### ISSUE THREAD

**Issue by janette** [GetDKAN/dkan#4260]
- Goals:
	- Mirror code from github project (this one) to the drupal.org DKAN project gitlab repo.
	- Allow users to see available updates from Drupal updates module.
	- Higher visibility from more Drupal users.
	- Benefit from d.o maintainer tools such as download stats.
	- Benefit from d.o security team work.
	- In the future, leverage Drupal autoupdates project.
	- Benefit from  Tugboat PR environments
	- CivicActions and our personnel benefit from D.o credit for all their work. (improves company standing/ improves personal brands)
- MVP (Phase 1):
	- Continue work as we do: Issues, PRs on github, nothing on d.o except releases.
	- Github action to mirror main branches from github to d.o.
	- Tasks:
		- Add `CONTRIBUTING.md` file.
		- Add license checker tooling to CI (Maybe https://github.com/dominikb/composer-license-checker ?)
		- Create github action to mirror github repo to d.o gitlab.
			- This should happen on push/merge to main branch (2.x at this time)
			- Should change the name in `composer.json` on push.
			- Example? https://git.drupalcode.org/project/drupalpod/-/blob/main/.github/workflows/mirror.yml?ref_type=heads
		- Update `README.md`
		- Determine release process for DKAN.
			- Strategy 1: Github releases are transformed into d.o releases somehow.
			- Strategy 2: Use D.O release API to create releases as github actions.
			- Strategy 3: Use D.O to make releases without any special scripting.
	- Notes:
		- D.O Project module docs: https://www.drupal.org/node/114618
		- How do release nodes get packaged into downloadable files? https://www.drupal.org/node/289662
- Phase 2:
	- All the things for D.O security policy approval. Add to this list.
		- Requirements for security policy approval https://www.drupal.org/node/1011698
		- `phpcs --standard=Drupal,DrupalPractice`
		- Probably consolidate external libraries as much as possible.
		- Might require shifting to DrupalCI.
			- Determine subset of tests that can/should/must be run on DrupalCI.
	- Mark `getdkan/dkan` as deprecated in favor of `drupal/dkan`.
	- Determine desirability/feasibility of turning past releases into d.o-based releases.
	- Determine desirability/feasibility of allowing d.o issues in addition to/replacing github issues.
		- Might be required for d.o security coverage.
		- Would require mirroring d.o issue branches to github, for CI.
		- Probably not practical at this phase.
	- Tasks:
		- TBD

**Comment by stefan-korn** [comment]
@janette : as DKAN 3.x is mentioned. Are there already more informations available about 3.x (like timline, features, BC)? There is no 3.x branch so far?

**Comment by janette** [comment]
Hi @stefan-korn, we are just beginning to plan for a 3.x branch. This will be an effort to reduce the complexity of the module and to follow more conventional Drupal class naming practices. Very early in the process, we are working on a roadmap that we will share once it is ready.

**Comment by dafeder** [comment]
@stefan-korn just to add here, I we are mostly talking about 3.x because we feel the very generic names of the Drupal sub-modules ("common", "harvest", "datastore") are problematic and we really want to both rename and combine several of them to make the namespacing follow Drupal conventions. Talking about releases on d.o have made this seem even more important. We are discussing a 3.x release because any such change would be breaking in the sense that namespaces would change and any downstream code would need to be updated.

**Comment by paul-m** [comment]
Let's move the DKAN 3.x stuff out of this issue and into its own meta/epic/etc.

**Comment by dmundra** [comment]
In addition to renaming the modules, we should add DKAN to the module titles because on the uninstall module page it is hard to find which ones are DKAN and which ones are not. The names 'Administration' and 'Search' I couldn't clearly see until I look at the description or looked at the modules on the 'Extend' page.

**Comment by paul-m** [comment]
3.x has its own label: https://github.com/GetDKAN/dkan/issues?q=is%3Aissue+is%3Aopen+label%3A3.x

And the module re-org has its own issue: https://github.com/GetDKAN/dkan/issues/3746

I think 3.x stuff isn't related here, so I'm moving that info to the other issue.

**Comment by paul-m** [comment]
Promoted this to an epic. We'll update the summary with github issues related to each task.

**Comment by swirtSJW** [comment]
Given that the Drupal issue queue is going to be moving to gitlab and that there are known methods for migrating from github to gitlab I suggest  making the full migration.  IF we still need circleCI on github to do testing then we let github be the mirror rather than the source of truth.  If Circle CI can run on gitlab, then all the better.
